x-pack/test/alerting_api_integration/security_and_spaces/tests/alerting/alerts.ts

```diff
@@ -33,55 +38,63 @@ export default function alertTests({ getService }: FtrProviderContext) {
       await es.indices.delete({ index: authorizationIndex });
     });
 
-    async function waitForTestIndexDoc(source: string, reference: string) {
-      return await retry.try(async () => {
-        const searchResult = await es.search({
-          index: esTestIndexName,
-          body: {
-            query: {
-              bool: {
-                must: [
-                  {
-                    term: {
-                      source,
-                    },
+    async function searchTestIndexDocs(source: string, reference: string) {
+      return await es.search({
+        index: esTestIndexName,
+        body: {
+          query: {
+            bool: {
+              must: [
+                {
+                  term: {
+                    source,
                   },
-                  {
-                    term: {
-                      reference,
-                    },
+                },
+                {
+                  term: {
+                    reference,
                   },
-                ],
-              },
+                },
+              ],
             },
           },
-        });
-        expect(searchResult.hits.total.value).to.eql(1);
-        return searchResult.hits.hits[0];
+        },
       });
     }
 
+    async function waitForTestIndexDocs(source: string, reference: string, numDocs: number = 1) {
+      return await retry.try(async () => {
+        const searchResult = await searchTestIndexDocs(source, reference);
+        expect(searchResult.hits.total.value).to.eql(numDocs);
+        return searchResult.hits.hits;
+      });
+    }
+
+    async function createIndexRecordAction(spaceId: string) {
+      const { body: createdAction } = await supertest
+        .post(`${getUrlPrefix(spaceId)}/api/action`)
```

Comment:
nit: just realized we can actually - **hopefully** - use the `.index` action to test this!  Unless there's some reason not to, maybe add a card to fix up later?

Reply:
I was thinking the same. Just did a check at how it differs and I have `test.index-record` also save the `config` and `secrets` which the alert doesn't have access to. It's a way to test to make sure things get decrypted and passed to the action type executor.